applications/SwimmingDEMApplication/custom_processes/codina2001_porosity_solution_and_body_force_process.h

```diff
@@ -0,0 +1,220 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Joaquin Gonzalez-Usua
+//
+//
+
+#ifndef KRATOS_CODINA2001_POROSITY_SOLUTION_AND_BODY_FORCE_PROCESS_H
+#define KRATOS_CODINA2001_POROSITY_SOLUTION_AND_BODY_FORCE_PROCESS_H
+
+// System includes
+#include <string>
+#include <iostream>
+
+// External includes
+
+// Project includes
+#include "includes/define.h"
+#include "processes/process.h"
+#include "includes/kratos_parameters.h"
+
+// Application includes
+
+
+namespace Kratos
+{
+///@addtogroup SwimmingDEMApplication
+///@{
+
+///@name Kratos Globals
+///@{
+
+///@}
+///@name Type Definitions
+///@{
+
+///@}
+///@name  Enum's
+///@{
+
+///@}
+///@name  Functions
+///@{
+
+///@}
+///@name Kratos Classes
+///@{
+
+class KRATOS_API(SWIMMING_DEM_APPLICATION) Codina2001PorositySolutionAndBodyForceProcess : public Process
+{
+public:
+    ///@name Type Definitions
+    ///@{
+
+    /// Node type (default is: Node<3>)
+    typedef Node NodeType;
+
+    /// Geometry type (using with given NodeType)
+    typedef Geometry<NodeType> GeometryType;
+
+    /// Definition of nodes container type, redefined from GeometryType
+    typedef Geometry<NodeType>::PointsArrayType NodesArrayType;
+
```

Comment:
lately we are using: `usign NodesArrayType = Geometry<NodeType>::PointsArrayType;`